user/test_main.py

```diff
@@ -0,0 +1,43 @@
+from fastapi.testclient import TestClient 
+from .main import app 
+
+client= TestClient(app) 
+
+data = {"name": "user5","email":"user5@gmail.com","password":"user5"}
+
+#Testcase for post api
+def test_create():
+    response=client.post("/create",json=data)
+    if response.status_code == 400:
+        assert response.json() == {"detail":"User already exists"}
+    else:
+        assert response.status_code == 201
+
+#Testcase for get api
+def test_retrieve():
+    response=client.get("/read",json=data)
+    if response.status_code == 400:
+        assert response.json() == "User not found"
+    else:
+        assert response.status_code == 200
+        assert response.json()
```

Comment:
We have to check for number of records returned from the DB are matching the number of entries. Including the data returned.